src/IntSet.java

```diff
@@ -0,0 +1,205 @@
+import java.util.ArrayList;
+import java.util.Collections;
+
+/*
+	Implement the immutable class intSet using an array to represent a set of integers in the range 1-1000.
+	It should support public methods like - 
+	
+	isMember(int x) - check whether x is a member of the set and return a boolean value
+	
+	size() - return the size of the set
+	
+	isSubSet(intSet s) - check whether s is a subset of the set
+	
+	getComplement()  - return the complement set, you can assume that 1..1000 is the universal set
+	
+	union(intSet s1, intSet s2) - return the union of s1 and s2
+	
+	You may use private helper methods. 
+*/
+
+public final class IntSet 
+{
+	private final int set[];
+	int  range=1000; //setting range for universal set;
+	private ArrayList <Integer> universal_set=new ArrayList<Integer>();
+	public static void main(String[] args)
+	{
+		getHelp();
+	}
+	
+	
+	public static void getHelp()
+	{
+		System.out.println("IntSet Operation Avaliable");
+		System.out.println("1. IntSet s = s.isMember(int x) - check whether x is a member of the set and return a boolean value");
+		System.out.println("2. IntSet s = s.size() - return the size of the set");
+		System.out.println("3. IntSet s = s.isSubSet(intSet s2) - check whether s is a subset of the set");
+		System.out.println("4. IntSet s = s.getComplement()  - return the complement set, 1..1000 is the universal set");
+		System.out.println("5. IntSet.union(intSet s1, intSet s2) - (Static method), return the union of s1 and s2");
+		System.out.println("6. IntSet.printSet(IntSet s) - print the set");
+		System.out.println("7. IntSet.setUniversalSet(int range) - set the range for universal set 1-range");
+	}
+	
+	
+	IntSet(int[] arr)
+	{
+		if(validateInput(arr)==true)
+		{
+			this.set=arr;
+		}
+		else
+		{
+			System.out.println("Invalid value for a set\n-> Duplicate item maybe present\n\tor\n-> Element out of Universal set are present\n   Universal set(1-1000)");
+			this.set=null;
+			System.exit(-1);
+		}
+	}
+	public boolean validateInput(int[] arr)
+	{
+		if(arr.length>range)
+		{
+			return false;
+		}
+		for(int value:arr)
+		{
+			if(value>range||value<1)
+			{
+				return false;
+			}
+		}
+		return !containDuplicateElements(arr);
+	}
+	
+	private boolean containDuplicateElements(int[] arr)
+	{
+		for(int loop_var=0;loop_var<arr.length;loop_var++)
+		{
```

Comment:
You could validate the out of range input here also to avoid one iteration.

Reply:
Resolved

Reply:
Still persist. I am talking about to move range validation at line 73, and remove line from 65 to 71.

Reply:
Merged in for loop.